servant-server/src/Servant/Server/Internal.hs

```diff
@@ -116,11 +118,10 @@ instance (KnownSymbol capture, FromText a, HasServer sublayout)
      a -> ServerT sublayout m
 
   route Proxy subserver =
-    DynamicRouter $ \ first ->
-      route (Proxy :: Proxy sublayout)
-            (case captured captureProxy first of
-               Nothing  -> return $ failWith NotFound
-               Just v   -> feedTo subserver v)
+    DynamicRouter $ \ first -> case captured captureProxy first of
+       Nothing  -> LeafRouter (\_ r -> r $ Fail err404)
+       Just v   -> route (Proxy :: Proxy sublayout) (feedTo subserver v)
+
```

Comment:
Why shouldn't captures be checked immediately? Currently by delaying it we let an endpoint with a capture followed by a method return a 405 (bad method) instead of a 404 if both the method and the url don't match. (Sorry, I don't remember this discussion!)

> then the `Capture` will be performed multiple times unless you left-factor by hand.

That does seem undesirable.